models/issue.go

```diff
@@ -426,6 +447,14 @@ func (issue *Issue) IsPoster(uid int64) bool {
 	return issue.OriginalAuthorID == 0 && issue.PosterID == uid
 }
 
+// CanSeeIssue returns true when a given user can view the issue.
+func (issue *Issue) CanSeeIssue(userID int64, repoPermission *Permission) bool {
+	if issue.IsPrivate {
+		return repoPermission.CanReadPrivateIssues() || issue.IsPoster(userID)
```

Comment:
I would also add users assigned to an issue, so an admin can assign specific maintainers to a secret issue

Reply:
well if you accidently asigned somebody you could also accidently make it public ...

Reply:
It would just add more granular control without needing everybody be an admin - I'll add it as optional TODO in the pull description so it wont get lost and if not implemented, a new issue can be created

Reply:
Yes, so that's why I don't think this should be by-default, it needs more thought into it. Nice to have as feature, but I'd prefer to not give a red button to the user to f*ck up.